test/parallel/test-zlib-from-string.js

```diff
@@ -54,30 +55,56 @@ const expectedBase64Gzip = 'H4sIAAAAAAAAA11RS05DMQy8yhzg6d2BPSAkJPZu4laWkjiN4' +
                            'mHo33kJO8xfkckmLjE5XMKBQ4gxIsfvCZ44doUThF2mcZq8q2' +
                            'sHnHNzRtagj5AQAA';
 
-zlib.deflate(inputString, common.mustCall((err, buffer) => {
-  zlib.inflate(buffer, common.mustCall((err, inflated) => {
-    assert.strictEqual(inflated.toString(), inputString);
-  }));
-}));
+test('properly deflate and inflate', async (t) => {
+  const { promise, resolve } = Promise.withResolvers();
+  zlib.deflate(inputString, (err, buffer) => {
+    assert.ifError(err);
+    zlib.inflate(buffer, (err, inflated) => {
+      assert.ifError(err);
+      assert.strictEqual(inflated.toString(), inputString);
+      resolve();
+    });
+  });
+  await promise;
```

Comment:
I love how `Promise.withResolvers()` helps making callback tests relatively concise, but I don't see how this is better than using `common.mustSucceed()`:
```suggestion
zlib.deflate(inputString, common.mustSucceed((buffer) => {
  zlib.inflate(buffer, common.mustSucceed((inflated) => {
    assert.strictEqual(inflated.toString(), inputString);
  }));
}));
```
Is there a reason why we should avoid relying on `common` utilities?